### PR TITLE
Make EventsService to use account from AuctionContext

### PR DESCRIPTION
**Pull request by DGarbar**
Refactor ExchangeService
RefactorTests

### CODE REVIEW

File: src/main/java/org/prebid/server/auction/ExchangeService.java [hunk 1]
```diff
@@ -130,7 +131,7 @@ public ExchangeService(BidderCatalog bidderCatalog, StoredResponseProcessor stor
         this.currencyService = currencyService;
         this.bidResponsePostProcessor = Objects.requireNonNull(bidResponsePostProcessor);
         this.privacyEnforcementService = Objects.requireNonNull(privacyEnforcementService);
-        this.eventsService = eventsService;
+        this.eventsService = Objects.requireNonNull(eventsService);
```
Comment (DGarbar) [inline src/main/java/org/prebid/server/auction/ExchangeService.java]: is there any reason why there wasn't null check ?

Comment (RossGoncharuk) [reply src/main/java/org/prebid/server/auction/ExchangeService.java]: I guess it just was missed, seems like EventsService is not a conditional bean therefore it always should be available. Either way, lets keep the null check - it won't harm, @rpanchyk right?

Comment (rpanchyk) [reply src/main/java/org/prebid/server/auction/ExchangeService.java]: Right, EventsService is mandatory. So, need to be checked for null. Thanks.

File: src/main/java/org/prebid/server/events/EventsService.java [hunk 2]
```diff
@@ -37,21 +28,10 @@ public Events createEvent(String bidId, String bidder) {
...
-    /**
-     * Returns false if account is not found or any exception occurred.
+     * This data is not critical, so returns false if null.
      */
-    private static Boolean accountFallback(Throwable exception) {
-        if (!(exception instanceof PreBidException)) {
-            logger.warn("Error occurred while fetching account", exception);
-        }
-        return false;
+    public Boolean isEventsEnabled(Account account) {
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/events/EventsService.java]: I think we don't this method anymore. We may check `EventsEnabled` directly in `EventsService`.

File: src/main/java/org/prebid/server/auction/ExchangeService.java [hunk 3]
```diff
@@ -185,8 +187,7 @@ public ExchangeService(BidderCatalog bidderCatalog, StoredResponseProcessor stor
                 .map(bidderResponses -> updateMetricsFromResponses(bidderResponses, publisherId))
                 .map(bidderResponses -> storedResponseProcessor.mergeWithBidderResponses(bidderResponses,
                         storedResponse, bidRequest.getImp()))
-                .compose(bidderResponses -> eventsService.isEventsEnabled(publisherId, timeout)
-                        .map(eventsEnabled -> Tuple2.of(bidderResponses, eventsEnabled)))
+                .map(bidderResponses -> Tuple2.of(bidderResponses, eventsService.isEventsEnabled(account)))
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/auction/ExchangeService.java]: Can we avoid tuple here?

File: src/main/java/org/prebid/server/auction/ExchangeService.java [hunk 4]
```diff
@@ -253,6 +253,15 @@ private static boolean isDebugEnabled(BidRequest bidRequest, ExtBidRequest extBi
         return extRequestPrebid != null && Objects.equals(extRequestPrebid.getDebug(), 1);
     }
 
+    /**
+     * Returns events enabled flag for the given account.
+     * <p>
+     * This data is not critical, so returns false if null.
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/auction/ExchangeService.java]: Not relevant anymore.

Comment (DGarbar) [reply src/main/java/org/prebid/server/auction/ExchangeService.java]: I can change to 
`Determines debug flag from {@link Account}` 
It would be ok ?

Comment (rpanchyk) [reply src/main/java/org/prebid/server/auction/ExchangeService.java]: Read once more - that's fine, never mind. Leave as is pls.

**Pull request merged**